packages/integration-tests/src/e2e-flow/e2e_flow.test.ts

```diff
@@ -0,0 +1,258 @@
+      type: 'branch',
+    };
+  });
+
+  after(async () => {
+    await cfnClient.send(
+      new DeleteStackCommand({
+        StackName: BackendIdentifierConversions.toStackName(
+          branchBackendIdentifier
+        ),
+      })
+    );
+    await fsp.rm(tempDir, { recursive: true });
+    // stop the npm proxy
+    await execa('npm', ['run', 'stop:npm-proxy'], { stdio: 'inherit' });
+  });
+
+  void it(`starting project`, async () => {
+    await execa(
+      packageManager.startsWith('yarn') ? 'yarn' : packageManager,
+      ['create', 'amplify', '--yes'],
+      {
+        cwd: tempDir,
+        stdio: 'inherit',
+      }
+    );
+
+    const amplifyPathPrefix = path.join(tempDir, 'amplify');
+
+    const pathPrefix = path.join(tempDir, 'amplify');
+
+    const files = await glob(path.join(amplifyPathPrefix, '**', '*'), {
+      // eslint-disable-next-line spellcheck/spell-checker
+      nodir: true,
+      windowsPathsNoEscape: true,
+      ignore: ['**/node_modules/**', '**/yarn.lock'],
+    });
+
+    const expectedAmplifyFiles = [
+      path.join('auth', 'resource.ts'),
+      'backend.ts',
+      path.join('data', 'resource.ts'),
+      'package.json',
+      'tsconfig.json',
+    ];
+
+    assert.deepStrictEqual(
+      files.sort(),
+      expectedAmplifyFiles.map((suffix) => path.join(pathPrefix, suffix))
+    );
+
+    if (packageManager === 'yarn-modern') {
+      await execa('yarn', ['config', 'set', 'nodeLinker', 'node-modules'], {
+        cwd: `${tempDir}/amplify`,
+        stdio: 'inherit',
+      });
+
+      await fsp.appendFile(
+        path.join(tempDir, '.yarnrc.yml'),
+        `pnpIgnorePatterns:\n  - ./nm-packages/**`
+      );
+      await execa('yarn', ['install'], {
+        cwd: tempDir,
+        stdin: 'inherit',
+      });
+      await execa('yarn', ['add', '@aws-amplify/backend'], {
+        cwd: `${tempDir}/amplify`,
+        stdio: 'inherit',
+      });
+    }
+
+    // assert that project compiles successfully
+    await execa(
+      packageManagerExecutable,
+      [
+        'tsc',
+        '--noEmit',
+        '--skipLibCheck',
+        // pointing the project arg to the amplify backend directory will use the tsconfig present in that directory
+        '--project',
+        amplifyPathPrefix,
+      ],
+      {
+        cwd: tempDir,
+        stdio: 'inherit',
+      }
+    );
+
+    if (packageManager.startsWith('yarn')) {
+      await execa('yarn', ['add', 'aws-cdk', 'aws-cdk-lib', 'constructs'], {
+        cwd: tempDir,
+        stdio: 'inherit',
+      });
+      if (packageManager === 'yarn-modern') {
+        await execa(
+          'yarn',
+          [
+            'add',
+            '-D',
+            'tsx',
+            'graphql',
+            'pluralize',
+            'zod',
+            '@aws-amplify/platform-core',
+            'esbuild',
+          ],
+          {
+            cwd: tempDir,
+            stdio: 'inherit',
+          }
+        );
```

Comment:
what happens if we don't do this? will brand new project fail deployment?
if yes then we should troubleshoot this further to figure out what's proper fix for this.

I.e. `<packageManagerExecutable> create amplify && <packageManagerExecutable> amplify sandbox (or pipeline deploy)`  should just work.

This looks like we might need to either do something with our dependencies  OR enchance create-amplify.

Btw. for the CDK part I suggest that you pull latest main - we made `create amplify` install CDK libs in customer's project recently.

Reply:
Simplified it [65df176](https://github.com/aws-amplify/amplify-backend/pull/901/commits/65df1765a01a3e3a0e7e004022ca9765b0fe8f17)

I'll make a follow-up PR to change create-amplify and pipeline-deploy.

Reply:
Done 038ac85e15

It should just work now